### ISSUE TITLE
Woo Installer: Premium plans with mapped domains are not prompted for plan upgrade

### ISSUE THREAD

**Issue by lsl** [Automattic/wp-calypso#59515]
Premium plan sites with mapped domains are treated as though they're able to be auto transferred instead of prompting for business upgrade.

https://user-images.githubusercontent.com/811776/147185569-3f6ad4d9-278c-4447-ad7b-606fa6c92d17.mp4

The above is caused by the way we're [filtering out](https://github.com/Automattic/wp-calypso/blob/1a7ef02d12b750ec6f1bdab83a96271c693f39d3/client/signup/steps/woocommerce-install/hooks/use-woop-handling/index.ts#L23) "missing a business plan" from the transfer blockers. I think this is so that the landing page doesn't block the user from entering the install flow if they need a plan upgrade, but then the confirm screen doesn't check if a site upgrade is required.

To fix we need [this check](https://github.com/Automattic/wp-calypso/pull/59517/files#diff-1b181ac4341b5a9dce199850a7a57e0768215bd3970b02b0174c29e4b2a6063eR73) to prevent auto transfer when an upgrade is required. Ideally, I think we should refactor the hook to make a distinction between transfer blockers and install flow blockers.

**Comment by lsl** [comment]
p1640223795136500-slack-C03TY6J1A @essleeung

**Comment by essleeung** [comment]
Another user in chat had the same behaviour - 21908068-hc

**Comment by lsl** [comment]
I've deployed D72103-code to remove the menu option. (From simple site users, atomic site users can still see the flow, all users can still enter the flow directly by url or with the woop site sticker.)

While testing my fix #59517 I noticed a secondary issue with premium sites after upgrading to business. The transfer starts and completes but the software status endpoint never issues `applied: true`. Since we're missing the same [process timeout](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/woocommerce-install/transfer/install-plugins.tsx#L18) that we have for the install flow, it just polls forever.

I decided at this point deploying D72103 was the safest option while I confirm and debug the secondary issue. If it is a real problem (and not just my sandbox / test sites) then dropping more users into an infinite poll screen with a background transfer completing sounded like a worse state.
